src/main/java/com/imsweb/mph/DefaultHematoDbUtilsProvider.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (C) 2016 Information Management Services, Inc.
+ */
+package com.imsweb.mph;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Pattern;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+import com.imsweb.mph.internal.HematoDbDTO;
+
+/**
+ * This is a default hemato db utils provider which uses seer-api to determine whether two morphologies are same primary, transform to or transform from according to
+ * hematopoietic and lymphoid neoplasm database.
+ */
+public class DefaultHematoDbUtilsProvider implements HematoDbUtilsProvider {
+
+    private Map<String, List<HematoDbDTO>> _samePrimaryDto = new HashMap<>();
+    private Map<String, List<HematoDbDTO>> _transformToDto = new HashMap<>();
+    private Map<String, List<HematoDbDTO>> _transformFromDto = new HashMap<>();
+    private static Pattern _MORPHOLOGY = Pattern.compile("^(\\d{4}/\\d)");
+
+    public DefaultHematoDbUtilsProvider() {
+        if (_samePrimaryDto.isEmpty()) {
```

Comment:
This check (and the two other similar checks) are not needed here; you know the collections are empty since you are creating a new instance of the object. In fact, I think it's better practice to leave the collection null when defining them, and assign an empty set in this constructor (instead of checking that they are empty).